### PR TITLE
Create templates from appliances

### PR DESCRIPTION
**Pull request by piotrpekala7**

### CODE REVIEW

File: src/app/components/project-map/new-template-dialog/new-template-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        fileReader.onloadend = () => {
+            let appliance = JSON.parse(fileReader.result as string);
+
+            if (appliance.docker) emulator = 'docker';
+            if (appliance.dynamips) emulator = 'dynamips';
+            if (appliance.iou) emulator = 'iou';
+            if (appliance.qemu) emulator = 'qemu';
+
+            const url = this.applianceService.getUploadPath(this.server, emulator, fileName);
```
Comment (piotrpekala7) [inline src/app/components/project-map/new-template-dialog/new-template-dialog.component.ts:95]: @ziajka still an error occurs during import

Comment (piotrpekala7) [reply src/app/components/project-map/new-template-dialog/new-template-dialog.component.ts:95]: any updates?

Comment (grossmj) [reply src/app/components/project-map/new-template-dialog/new-template-dialog.component.ts:95]: Indeed, there is an error when importing a .gns3a file.

Screenshot from 2020-06-11 16-37-33

Unfortunately, I don't see why you get this error... :(

Comment (piotrpekala7) [reply src/app/components/project-map/new-template-dialog/new-template-dialog.component.ts:95]: @grossmj Can u confirm that this endpoint `http://${server.host}:${server.port}/v2/${emulator}/images/${filename}` is correct?

Comment (grossmj) [reply src/app/components/project-map/new-template-dialog/new-template-dialog.component.ts:95]: Looks like you are missing `/compute` after `/v2`.

Here is a request example:

```
curl -v -H "Content-Type:application/octet-stream" "http://localhost:3080/v2/compute/dynamips/images/c7200-adventerprisek9-mz.153-3.XB12.bin" --data-binary "@c7200-adventerprisek9-mz.153-3.XB12.bin"
```

**Pull request merged**